db/version_builder.cc

```diff
@@ -233,6 +231,19 @@ class VersionBuilder::Rep {
     }
   }
 
+  bool CheckConsistencyForNumLevels() {
+    // Make sure there are no files on or beyond num_levels().
+    auto level_iter = levels_.lower_bound(base_vstorage_->num_levels());
+    while (level_iter != levels_.end()) {
+      if (level_iter->second.added_files.size() != 0) {
+        return false;
+      }
+      // Don't need to check deleted files.
+      level_iter = levels_.erase(level_iter);
```

Comment:
I'm not a big fan of deleting entries from a data structure while iterating it, no matter it is supported or not, because it creates a portable issue if someone tries to swap a data structure. I suggest we do it in two paths, or not removing these empty levels at all.